src/lib.rs

```diff
@@ -399,6 +401,63 @@ impl<'a, 'b> Builder<'a, 'b> {
         self
     }
 
+    /// The permissions to create the tempfile with.
+    /// This allows to them differ from the default mode of `0o600` on Unix.
+    ///
+    /// # Security
+    ///
+    /// By default, the permissions of tempfiles on unix are set for it to be
+    /// readable and writable by the owner only, yielding the greatest amount
+    /// of security.
+    /// As this method allows to widen the permissions, security would be
+    /// reduced in such cases.
+    ///
+    /// # Platform Notes
+    /// ## Unix
+    ///
+    /// The actual permission bits set on the tempfile will be affected by the
+    /// `umask` applied by the underlying `open` syscall.
+    ///
+    /// ## Windows and others
+    ///
+    /// This setting is ignored.
+    ///
+    /// # Limitations
+    ///
+    /// Permissions for directories aren't currently set even though it would
+    /// be possible on Unix systems.
```

Comment:
We need to implement this before we merge this patch, unfortunately. This builder is used to create temporary directories as well, and I don't really trust users to read the docs.

Reply:
See the discussion in #61.

Reply:
I took a look at this and saw that `std::fs::create_dir()` is used to create the directory, I don't think the standard library has a way to handle this.
Did you have an idea on how to do this? Maybe a `rustix` `chmod` call?

Reply:
I think we'll need to use `rustix::fs::mkdir`.

Reply:
This comment is now out of date.